includes/frontend/class-wcmp-checkout.php

```diff
@@ -564,6 +563,43 @@ private function shouldShowDeliveryOptions(): bool
         return apply_filters("wc_myparcel_show_delivery_options", $showDeliveryOptions);
     }
 
+    /**
+     * @param  string $carrierName
+     *
+     * @return bool
+     */
+    private function shouldShowSameDayDelivery(string $carrierName): bool
+    {
+        $settingCollection    = WCMYPA()->setting_collection->where('carrier', $carrierName);
+        $carrierIsActive      = (bool) $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DELIVERY_ENABLED);
+        $dropOffDays          = $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DROP_OFF_DAYS);
+        $sameDayFromSettings  = (bool) $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_SAME_DAY_DELIVERY);
+        $isInSameDayTimeSlot  = $this->isInSameDayTimeSlot($carrierName);
+        $tomorrowIsDropOffDay = in_array(date('N'), $dropOffDays, true);
+        $noDropOffDelay       = "0" === $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DROP_OFF_DELAY);
+
+        return $carrierIsActive && $sameDayFromSettings && $tomorrowIsDropOffDay && $noDropOffDelay && $isInSameDayTimeSlot;
```

Comment:
Misschien wordt het beter als je bijvoorbeeld `$tomorrowIsDropOffDay && $noDropOffDelay && $isInSameDayTimeSlot` naar een losse functie trekt. Dan zou ik ook eigenlijk snel willen dat dat niet meer in deze class staat, maar in iets als een `SameDayDeliveryService` oid